File: 57-exercises-for-programmers/c-sharp/exercise_03.cs
// Create a program that prompts for a quote and an author. Display the
// quotation and author as shown in the example output.

public class Exercise3 {
	string Author {get; set; }
	string Quote { get; set; }

	public Exercise3 () {
		PromptQuote();
		PromptAuthor();
		PrintQuote();
	}

	void PromptQuote() {
		System.Console.Write("What's the quote? ");
		Quote = System.Console.ReadLine();
	}

	void PromptAuthor() {
		System.Console.Write("Who's the author? ");
		Author = System.Console.ReadLine();
	}

	void PrintQuote() {
		System.Console.WriteLine();
		System.Console.WriteLine("===Output===");
		System.Console.WriteLine("{0} says, \"{1}\"", Author, Quote);
	}
}

File: 57-exercises-for-programmers/rust/src/exercise_03.rs
// Create a program that prompts for a quote and an author. Display the
// quotation and author as shown in the example output.

pub mod exercise_3 {

    struct Quotation {
        author : String,
        quote : String,
    }

    pub fn run() {
        display_quotation(prompt_quotation())
    }

    fn prompt_quotation() -> Quotation {
        use std::io::{stdin, stdout, Write};

        let mut author = String::new();
        let mut quote = String::new();

        print!("What is the quote? ");
        stdout().flush().unwrap();

        stdin().read_line(&mut quote).expect("Error: unable to read quote.");

        print!("What said it? ");
        stdout().flush().unwrap();

        stdin().read_line(&mut author).expect("Error: unable to read author.");

        let author = author.trim().to_string();
        let quote = quote.trim().to_string();

        Quotation { author, quote }
    }

    fn display_quotation (quotation : Quotation) {
        println!("{} says, \"{}\"", quotation.author, quotation.quote);
    }
}
